runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -94,19 +96,53 @@ impl IntoVMError for wasmer::RuntimeError {
         let error_msg = self.message();
         let trap_code = self.clone().to_trap();
         match self.downcast::<VMLogicError>() {
-            Ok(e) => (&e).into(),
-            _ => {
-                if let Some(trap_code) = trap_code {
-                    // A trap
-                    VMError::FunctionCallError(FunctionCallError::Wasmer1Trap(
-                        trap_code.to_string(),
-                    ))
-                } else {
-                    // A general error
-                    VMError::FunctionCallError(FunctionCallError::WasmerRuntimeError(error_msg))
-                }
-            }
+            Ok(e) => return (&e).into(),
+            _ => {}
         }
+        let trap_code = trap_code.unwrap_or_else(|| panic!("Unknown error: {}", error_msg));
```

Comment:
Why does it make sense to panic here?

Reply:
Because we cannot distinguish if this error is deterministic or not.

Reply:
We should leave a comment that says, that if this happens that it is a bug in Wasmer, since their errors should have an exhaustive coverage, without "unknown errors".

Reply:
Done